### ISSUE TITLE
Dropbox and Xapp Status Applet problems after upgrade

### ISSUE THREAD

**Issue by mfreeman72** [linuxmint/linuxmint#189]
This maybe should have been two separate items, so I apologize if that's the case...

I upgraded in-place from Linux Mint Cinnamon 64-bit 19.2 to 19.3, and have a couple problems…

First, every time I login, Dropbox now tells me that I need to install their proprietary daemon, even though it’s already running and if I go ahead and download it, it runs a second instance of it in my notification tray, so I have to quit one. Dropbox functioned perfectly fine before the upgrade.

Second, and also every time I login, I’m getting a triangle with an exclamation point in it on my notification tray. When I hover my mouse over it, it says, “Add the XApp Status applet”. However, that has already been added, and the Applets Settings shows that it is enabled. When I click on this triangle icon, the System Reports tool comes up, but says “No problems were detected.” Could this be some sort of race condition between the normal Systray applet and Xapp Status applet? The Systray starts up, and the system doesn’t detect the Xapp Status applet yet, so it throws the error just before it runs? Just a thought. I've tried both logging out and back in as well as full restarts. On one of login attempts (after a full restart in this case), it did start without either of these issues happening, but I can't repeat that result in any subsequent logins/restarts.

System info:
[code]
System:    Host: mike-Satellite-C75D-B Kernel: 5.3.0-24-generic x86_64 bits: 64 compiler: gcc 
           v: 7.4.0 Desktop: Cinnamon 4.4.6 wm: muffin dm: LightDM Distro: Linux Mint 19.3 Tricia 
           base: Ubuntu 18.04 bionic 
Machine:   Type: Laptop System: TOSHIBA product: Satellite C75D-B v: PSCLEU-009011 
           serial: <filter> 
           Mobo: TOSHIBA model: Portable PC v: MP serial: <filter> UEFI [Legacy]: Insyde v: 1.40 
           date: 11/03/2014 
Battery:   ID-1: BAT0 charge: 34.6 Wh condition: 34.6/48.6 Wh (71%) volts: 10.8/10.8 
           model: PA5109U-1BRS serial: <filter> status: Full 
           Device-1: hidpp_battery_0 model: Logitech Wireless Mouse M510 serial: <filter> 
           charge: 55% status: Discharging 
CPU:       Topology: Quad Core model: AMD A8-6410 APU with AMD Radeon R5 Graphics bits: 64 
           type: MCP arch: Puma rev: 1 L2 cache: 2048 KiB 
           flags: lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm bogomips: 15969 
           Speed: 2229 MHz min/max: 1000/2000 MHz Core speeds (MHz): 1: 2283 2: 2248 3: 2255 
           4: 2303 
Graphics:  Device-1: AMD Mullins [Radeon R4/R5 Graphics] vendor: Toshiba America Info Systems 
           driver: radeon v: kernel bus ID: 00:01.0 chip ID: 1002:9851 
           Display: x11 server: X.Org 1.20.4 driver: ati,radeon unloaded: fbdev,modesetting,vesa 
           resolution: 1600x900~60Hz 
           OpenGL: renderer: AMD MULLINS (DRM 2.50.0 5.3.0-24-generic LLVM 8.0.0) 
           v: 4.5 Mesa 19.0.8 direct render: Yes 
Audio:     Device-1: AMD Kabini HDMI/DP Audio vendor: Toshiba America Info Systems 
           driver: snd_hda_intel v: kernel bus ID: 00:01.1 chip ID: 1002:9840 
           Device-2: AMD FCH Azalia vendor: Toshiba America Info Systems driver: snd_hda_intel 
           v: kernel bus ID: 00:14.2 chip ID: 1022:780d 
           Sound Server: ALSA v: k5.3.0-24-generic 
Network:   Device-1: Realtek RTL810xE PCI Express Fast Ethernet 
           vendor: Toshiba America Info Systems driver: r8169 v: kernel port: 2000 bus ID: 01:00.0 
           chip ID: 10ec:8136 
           IF: enp1s0 state: down mac: <filter> 
           Device-2: Qualcomm Atheros AR9485 Wireless Network Adapter vendor: Lite-On 
           driver: ath9k v: kernel port: 2000 bus ID: 05:00.0 chip ID: 168c:0032 
           IF: wlp5s0 state: up mac: <filter> 
Drives:    Local Storage: total: 698.64 GiB used: 584.37 GiB (83.6%) 
           ID-1: /dev/sda vendor: Toshiba model: MQ01ABD075 size: 698.64 GiB speed: 3.0 Gb/s 
           serial: <filter> 
Partition: ID-1: / size: 681.78 GiB used: 584.37 GiB (85.7%) fs: ext4 dev: /dev/dm-0 
           ID-2: swap-1 size: 4.97 GiB used: 0 KiB (0.0%) fs: swap dev: /dev/dm-1 
USB:       Hub: 1-0:1 info: Full speed (or root) Hub ports: 2 rev: 2.0 chip ID: 1d6b:0002 
           Hub: 1-1:2 info: Advanced Micro Devices ports: 4 rev: 2.0 chip ID: 0438:7900 
           Device-1: 1-1.1:3 info: Logitech Unifying Receiver type: Keyboard,Mouse,HID 
           driver: logitech-djreceiver,usbhid rev: 2.0 chip ID: 046d:c52b 
           Hub: 2-0:1 info: Full speed (or root) Hub ports: 2 rev: 2.0 chip ID: 1d6b:0002 
           Hub: 2-1:2 info: Advanced Micro Devices ports: 4 rev: 2.0 chip ID: 0438:7900 
           Device-2: 2-1.2:3 info: Chicony type: Video driver: uvcvideo rev: 2.0 
           chip ID: 04f2:b448 
           Hub: 3-0:1 info: Full speed (or root) Hub ports: 2 rev: 2.0 chip ID: 1d6b:0002 
           Hub: 4-0:1 info: Full speed (or root) Hub ports: 2 rev: 3.0 chip ID: 1d6b:0003 
Sensors:   System Temperatures: cpu: 66.0 C mobo: N/A gpu: radeon temp: 64 C 
           Fan Speeds (RPM): N/A 
Repos:     No active apt repos in: /etc/apt/sources.list 
           Active apt repos in: /etc/apt/sources.list.d/adamreichold-qpdfview-dailydeb-bionic.list 
           1: deb http: //ppa.launchpad.net/adamreichold/qpdfview-dailydeb/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/additional-repositories.list 
           1: deb https: //dl.winehq.org/wine-builds/ubuntu/ bionic main
           2: deb http: //mirrors.dotsrc.org/getdeb/ubuntu zesty-getdeb apps
           3: deb http: //mirrors.dotsrc.org/getdeb/ubuntu zesty-getdeb games
           4: deb https: //apt.iteas.at/iteas bionic main
           5: deb [arch=amd64] https: //download.virtualbox.org/virtualbox/debian bionic contrib
           6: deb https: //dl.bintray.com/celestia/releases-deb bionic universe
           Active apt repos in: /etc/apt/sources.list.d/arunsivaraman-gpuviewer-bionic.list 
           1: deb http: //ppa.launchpad.net/arunsivaraman/gpuviewer/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/atareao-atareao-bionic.list 
           1: deb http: //ppa.launchpad.net/atareao/atareao/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/atareao-nemo-extensions-bionic.list 
           1: deb http: //ppa.launchpad.net/atareao/nemo-extensions/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/bartbes-love-stable-bionic.list 
           1: deb http: //ppa.launchpad.net/bartbes/love-stable/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/cybolic-ppa-bionic.list 
           1: deb http: //ppa.launchpad.net/cybolic/ppa/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/fossfreedom-rhythmbox-plugins-bionic.list 
           1: deb http: //ppa.launchpad.net/fossfreedom/rhythmbox-plugins/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/gerardpuig-ppa-bionic.list 
           1: deb http: //ppa.launchpad.net/gerardpuig/ppa/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/google-chrome.list 
           1: deb [arch=amd64] http: //dl.google.com/linux/chrome/deb/ stable main
           Active apt repos in: /etc/apt/sources.list.d/google-earth-pro.list 
           1: deb http: //dl.google.com/linux/earth/deb/ stable main
           Active apt repos in: /etc/apt/sources.list.d/google-musicmanager.list 
           1: deb [arch=amd64] http: //dl.google.com/linux/musicmanager/deb/ stable main
           Active apt repos in: /etc/apt/sources.list.d/google-talkplugin.list 
           1: deb [arch=amd64] http: //dl.google.com/linux/talkplugin/deb/ stable main
           Active apt repos in: /etc/apt/sources.list.d/helfrich-ppa-bionic.list 
           1: deb http: //ppa.launchpad.net/helfrich/ppa/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/jgeboski.list 
           1: deb http: //download.opensuse.org/repositories/home:/jgeboski/xUbuntu_18.04 ./
           Active apt repos in: /etc/apt/sources.list.d/langdalepl-pidgin-indicator-bionic.list 
           1: deb http: //ppa.launchpad.net/langdalepl/pidgin-indicator/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/libreoffice-libreoffice-prereleases-bionic.list 
           1: deb http: //ppa.launchpad.net/libreoffice/libreoffice-prereleases/ubuntu bionic main
           No active apt repos in: /etc/apt/sources.list.d/libreoffice-ppa-bionic.list 
           Active apt repos in: /etc/apt/sources.list.d/lightzone-team-lightzone-bionic.list 
           1: deb http: //ppa.launchpad.net/lightzone-team/lightzone/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/linuxuprising-java-bionic.list 
           1: deb http: //ppa.launchpad.net/linuxuprising/java/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/lutris-team-lutris-bionic.list 
           1: deb http: //ppa.launchpad.net/lutris-team/lutris/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/megasync.list 
           1: deb https: //mega.nz/linux/MEGAsync/xUbuntu_18.04/ ./
           Active apt repos in: /etc/apt/sources.list.d/morphis-anbox-support-bionic.list 
           1: deb http: //ppa.launchpad.net/morphis/anbox-support/ubuntu bionic main
           2: deb-src http: //ppa.launchpad.net/morphis/anbox-support/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/mscore-ubuntu-mscore3-stable-bionic.list 
           1: deb http: //ppa.launchpad.net/mscore-ubuntu/mscore3-stable/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/ngld-knossos-bionic.list 
           1: deb http: //ppa.launchpad.net/ngld/knossos/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/nilarimogard-webupd8-bionic.list 
           1: deb http: //ppa.launchpad.net/nilarimogard/webupd8/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/obsproject-obs-studio-bionic.list 
           1: deb http: //ppa.launchpad.net/obsproject/obs-studio/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list 
           1: deb http: //packages.linuxmint.com tricia main upstream import backport #id:linuxmint_main
           2: deb http: //archive.ubuntu.com/ubuntu bionic main restricted universe multiverse
           3: deb http: //archive.ubuntu.com/ubuntu bionic-updates main restricted universe multiverse
           4: deb http: //archive.ubuntu.com/ubuntu bionic-backports main restricted universe multiverse
           5: deb http: //security.ubuntu.com/ubuntu/ bionic-security main restricted universe multiverse
           6: deb http: //archive.canonical.com/ubuntu/ bionic partner
           Active apt repos in: /etc/apt/sources.list.d/oguzhaninan-stacer-bionic.list 
           1: deb http: //ppa.launchpad.net/oguzhaninan/stacer/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/olive-editor-olive-editor-bionic.list 
           1: deb http: //ppa.launchpad.net/olive-editor/olive-editor/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/openshot_developers-libopenshot-daily-bionic.list 
           1: deb http: //ppa.launchpad.net/openshot.developers/libopenshot-daily/ubuntu bionic main
           No active apt repos in: /etc/apt/sources.list.d/openshot_developers-ppa-bionic.list 
           Active apt repos in: /etc/apt/sources.list.d/playonlinux.list 
           1: deb http: //deb.playonlinux.com/ bionic main
           Active apt repos in: /etc/apt/sources.list.d/saiarcot895-flightgear-bionic.list 
           1: deb http: //ppa.launchpad.net/saiarcot895/flightgear/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/scribus-ppa-bionic.list 
           1: deb http: //ppa.launchpad.net/scribus/ppa/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/skype-stable.list 
           1: deb [arch=amd64] https: //repo.skype.com/deb stable main
           Active apt repos in: /etc/apt/sources.list.d/spotify.list 
           1: deb http: //repository.spotify.com stable non-free
           Active apt repos in: /etc/apt/sources.list.d/stebbins-handbrake-releases-bionic.list 
           1: deb http: //ppa.launchpad.net/stebbins/handbrake-releases/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/stjr-srb2-bionic.list 
           1: deb http: //ppa.launchpad.net/stjr/srb2/ubuntu bionic main
           2: deb-src http: //ppa.launchpad.net/stjr/srb2/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/stk-dev-bionic.list 
           1: deb http: //ppa.launchpad.net/stk/dev/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/team-xbmc-ppa-bionic.list 
           1: deb http: //ppa.launchpad.net/team-xbmc/ppa/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/team-xbmc-unstable-bionic.list 
           1: deb http: //ppa.launchpad.net/team-xbmc/unstable/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/teejee2008-ppa-bionic.list 
           1: deb http: //ppa.launchpad.net/teejee2008/ppa/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/ubuntuhandbook1-audacity-bionic.list 
           1: deb http: //ppa.launchpad.net/ubuntuhandbook1/audacity/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/ubuntuhandbook1-dvdstyler-bionic.list 
           1: deb http: //ppa.launchpad.net/ubuntuhandbook1/dvdstyler/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/webupd8team-brackets-bionic.list 
           1: deb http: //ppa.launchpad.net/webupd8team/brackets/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/webupd8team-java-bionic.list 
           1: deb http: //ppa.launchpad.net/webupd8team/java/ubuntu bionic main
Info:      Processes: 222 Uptime: 1h 04m Memory: 4.79 GiB used: 1.38 GiB (28.9%) Init: systemd 
           v: 237 runlevel: 5 Compilers: gcc: 7.4.0 alt: 7 Client: Unknown python3.6 client 
           inxi: 3.0.32 
[/code]

**Comment by mtwebster** [comment]
The xapp alert thing I think is a known issue, the dropbox problem I would first maybe check out permissions/ownership in your home folder - there's a .dropbox, .dropbox-dist - check those - they should be owned by you.  Pretty much everything in your home folder, recursively, should be owned by you.  The exception may be items in the .dbus folder.

I'd also try completely removing dropbox and re-installing it:
```
sudo apt-get remove --purge dropbox
```
This will probably uninstall nemo-dropbox as well, not a big deal.

See if the uninstall removed those two folders I mentioned above.  If they're still there, delete or rename them.

Reboot or log out/in and then reinstall - `sudo apt-get install nemo-dropbox dropbox` - you'll probably
have to log back in again, etc.. but hopefully for the last time.

**Comment by mfreeman72** [comment]
@mtwebster I tried everything you suggested, but the problem is still occuring. Interestingly, if I remove Dropbox from my Startup Applications, not only does Dropbox not start at all (which is obviously expected), but the alert about the Xapp Status Applet goes away as well! I can start Dropbox manually after Cinnamon loads, and everything works perfectly. But if I enable Dropbox in my Startup Applications, both problems return on the next login/reboot.

**Comment by mtwebster** [comment]
Okay, one last thing then:
Look in `~/.config/autostart` and see if there is more than one dropbox desktop file there (or maybe under a different name).

It seems as though the startup settings module will avoid overwriting a pre-existing desktop file the first time a program is added.  So if, for some reason, dropbox.desktop was already there (added by the dropbox daemon itself via `dropbox autostart y`), it will add one called 'dropbox-1.desktop.

I don't know if this is the issue, but it sounds promising, as you said the behavior changed when you disabled it.  Try removing any entries from that folder, and try adding it again via the startup settings.

**Comment by mfreeman72** [comment]
There was only one dropbox.desktop in ~/.config/autostart. I removed it, and re-added it as you suggested. After a restart, both problems are back again.

I can't shake this feeling that the two problems are related, especially since this didn't happen until the upgrade to 19.3, and since keeping Dropbox from autostarting fixes the Xapp Status Applet problem as well. I could very well be wrong about that, though.

I'm going to try creating a new user and see if it happens there as well. I'll report back after I've done that.

**Comment by mfreeman72** [comment]
@mtwebster Well, I created a new user, and these problems did not show up there. So I came back to my original user account, and I got it as close to a stock out-of-the-box user account as I could by disabling all additional applets, extensions, startup applications, etc. that don't come with a new user account, except I kept Dropbox in place. I rebooted, and everything was fine - no Dropbox download, and no Xapp Status Applet error. I began to add everything back in one at a time followed by logout/logins. After all that, I now have everything exactly where it was before, I've done a few logins/reboots, and now the problem is no longer there. I have no idea what this was. But it's gone now, hopefully. Thanks for the help!
